### PR TITLE
add rpc framework based on sofa-rpc

### PR DESCRIPTION
**Pull request by javeme**
Co-authored-by: Jermy Li <lizhangmei@baidu.com>
Co-authored-by: xuliguov5 <xuliguo@baidu.com>

### CODE REVIEW

File: src/main/java/com/baidu/hugegraph/rpc/RpcConsumerConfig.java [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+
+    @Override
+    public <T> T serviceProxy(String graph, String interfaceId) {
+        ConsumerConfig<T> config = this.consumerConfig(graph, interfaceId);
+        return config.refer();
+    }
+
+    @Override
+    public <T> T serviceProxy(String interfaceId) {
+        ConsumerConfig<T> config = this.consumerConfig(null, interfaceId);
```
Comment (houzhizhen) [inline src/main/java/com/baidu/hugegraph/rpc/RpcConsumerConfig.java]: return serviceProxy(null, interfaceId) ?

File: src/main/java/com/baidu/hugegraph/rpc/RpcServer.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    private final ServerConfig serverConfig;
+
+    public RpcServer(HugeConfig conf) {
+        RpcCommonConfig.initRpcConfigs(conf);
+        this.conf = conf;
+        this.serverConfig = new ServerConfig();
+        this.serverConfig.setProtocol(conf.get(RpcOptions.RPC_PROTOCOL))
+                         .setHost(conf.get(RpcOptions.RPC_SERVER_HOST))
+                         .setPort(conf.get(RpcOptions.RPC_SERVER_PORT))
+                         .setDaemon(false);
```
Comment (houzhizhen) [inline src/main/java/com/baidu/hugegraph/rpc/RpcServer.java]: How to listen on a random port? There are a few of workers on a computer, and they can't listen on same port.

Comment (javeme) [reply src/main/java/com/baidu/hugegraph/rpc/RpcServer.java]: maybe we can use sofa  option`server.adaptive.port`, or 0 means random port.

seems there is bug using random port: https://github.com/sofastack/sofa-bolt/issues/196 , and fixed bolt v1.6.1

**Review by coderzc — DISMISSED** [review]

**Review by houzhizhen — DISMISSED** [review]

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+language: java
+
+jdk:
+  - openjdk8
```
Comment (Linary) [inline .travis.yml]: jdk:
  - oraclejdk8

dist: trusty

**Review by Linary — CHANGES_REQUESTED** [review]

**Review by Linary — DISMISSED** [review]

**Review by coderzc — DISMISSED** [review]

**Review by Linary — APPROVED** [review]

**Review by zhoney — APPROVED** [review]

**Pull request merged**